composer/core/state.py

```diff
@@ -540,14 +547,117 @@ def state_dict(self) -> Dict[str, Any]:
 
         return state_dict
 
-    def load_model_state(self, state_dict: Dict[str, Any], strict: bool):
+    def _apply_required_algorithms(
+        self,
+        state: Dict[str, Any],
```

Comment:
Any chance we could call this `state_dict`? I think that will add clarity because it is a bit more specific and `state` is already the name of this module and the name of the class